src/main/java/com/google/sps/data/DeliverySystem.java

```diff
@@ -26,16 +26,74 @@
 import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
+import java.util.List;
 
 /** 
  * Class that assigns orders and creates delivery journeys for delivery slots.
  */
 public class DeliverySystem {
+  private final Integer MAX_WAYPOINTS = 25;
+  private PathFinder pathFinder;
+
+  public DeliverySystem(PathFinder pathFinder) {
+    this.pathFinder = pathFinder;
+  }
+
   /**
    * Finds the order in which points must be visited such that the journey lasts minimum time and it
    * delivers all orders assigned in findOrdersForDeliveryRequest.
    */
-  public ArrayList<Point> findOptimalDeliveryJourneyForDeliverySlot(DeliverySlot deliverySlot) throws ApiException, IOException, InterruptedException {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public ArrayList<Point> findOptimalDeliveryJourneyForDeliverySlot(DeliverySlot deliverySlot) throws ApiException, BadRequestException, DataNotFoundException, IOException, InterruptedException {
```

Comment:
You should always try to minimize the number of public methods. After this PR, this public method does not provide value over findOrdersForDeliveryRequest (other than saving you to type a few characters, which is not a good reason).

`findOrdersForDeliveryRequest` is also dead-on-arrival method (only called in tests), which is an antipattern.

You can split them if you find a good splitting point (one method creates a journey, the other takes the journey you just created and "optimizes" it). I would probably just keep one of them though which does everything.

Reply:
I removed `findOptimalDeliveryJourneyForDeliverySlot` and renamed `findOrdersForDeliveryRequest` to `createJourneyForDeliverySlot`.